### PR TITLE
feat(HACBS-2359): do not rely on public key from EC config

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-2359

### CODE REVIEW

**Comment by lcarva** [comment]
It looks like a lot of changes but this is basically the diff across all the pipelines:
```diff
diff 0.19/README.md 0.20/README.md
14a15
> | enterpriseContractPublicKey | Public key to use for validation by the enterprise contract | No | k8s://openshift-pipelines/public-key |
22a24,26
> 
> ## Changes since 0.19
> - add new enterpriseContractPublicKey parameter
diff 0.19/release.yaml 0.20/release.yaml
33a34,37
>     - name: enterpriseContractPublicKey
>       type: string
>       description: Public key to use for validation by the enterprise contract
>       default: k8s://openshift-pipelines/public-key
162a167,168
>         - name: PUBLIC_KEY
>           value: $(params.enterpriseContractPublicKey)
Common subdirectories: 0.19/samples and 0.20/samples
Common subdirectories: 0.19/tests and 0.20/tests
```

Not sure if a new version is needed. I could just add the changes to the latest version if that's preferred.

**Comment by johnbieren** [comment]
> It looks like a lot of changes but this is basically the diff across all the pipelines:
> 
> ```diff
> diff 0.19/README.md 0.20/README.md
> 14a15
> > | enterpriseContractPublicKey | Public key to use for validation by the enterprise contract | No | k8s://openshift-pipelines/public-key |
> 22a24,26
> > 
> > ## Changes since 0.19
> > - add new enterpriseContractPublicKey parameter
> diff 0.19/release.yaml 0.20/release.yaml
> 33a34,37
> >     - name: enterpriseContractPublicKey
> >       type: string
> >       description: Public key to use for validation by the enterprise contract
> >       default: k8s://openshift-pipelines/public-key
> 162a167,168
> >         - name: PUBLIC_KEY
> >           value: $(params.enterpriseContractPublicKey)
> Common subdirectories: 0.19/samples and 0.20/samples
> Common subdirectories: 0.19/tests and 0.20/tests
> ```
> 
> Not sure if a new version is needed. I could just add the changes to the latest version if that's preferred.

Our convention for our READMEs is to set `Optional` to `Yes`, not `No` if there is a default value. Can you adjust?

**Comment by johnbieren** [comment]
If the currently used value is the same as the default that is being provided here, maybe we don't need new versions? @redhat-appstudio/release-team wdyt

**Comment by scoheb** [comment]
> If the currently used value is the same as the default that is being provided here, maybe we don't need new versions? @redhat-appstudio/release-team wdyt

The default value is what we would expect from all of our EC policies that we managed.

I am fine with adding this to existing versions

**Comment by lcarva** [comment]
>  Our convention for our READMEs is to set Optional to Yes, not No if there is a default value. Can you adjust?

Done! Thanks

**Comment by lcarva** [comment]
> The default value is what we would expect from all of our EC policies that we manage.
> 
> I am fine with adding this to existing versions

Yeah, there shouldn't be any change in usage or behavior. Updated.

File: catalog/pipeline/deploy-release/0.7/README.md [hunk 1]
```diff
@@ -12,10 +12,14 @@ Tekton pipeline to verify Snapshot prior to Deployment
 | releasestrategy | The namespaced name (namespace/name) of the releaseStrategy | No | - |
 | snapshot | The namespaced name (namespace/name) of the snapshot | No | - |
 | enterpriseContractPolicy | JSON representation of the policy to be applied when validating the enterprise contract | No | - |
+| enterpriseContractPublicKey | Public key to use for validation by the enterprise contract | Yes | k8s://openshift-pipelines/public-key |
 | verify_ec_task_git_url | The git repo url of the verify-enterprise-contract task | No | - |
 | verify_ec_task_git_revision | The git repo revision the verify-enterprise-contract task | No | - |
 | verify_ec_task_git_pathInRepo | The location of the verify-enterprise-contract task in its repo | No | - |
 
+## Changes since 0.7
```
Comment (johnbieren) [inline catalog/pipeline/deploy-release/0.7/README.md:20]: Can you remove this and put L21 in the section below (so move it two lines down)? Same for all of them

**Review by johnbieren — COMMENTED** [review]
One comment but otherwise lgtm

File: catalog/pipeline/deploy-release/0.7/README.md (hunk 1 above)
Comment (lcarva) [reply catalog/pipeline/deploy-release/0.7/README.md]: My apologies. That was a careless mistake. Fixed.

**Review by johnbieren — APPROVED** [review]

**Pull request merged**